agent/cache/cache.go

```diff
@@ -829,7 +804,6 @@ func (c *Cache) Prepopulate(t string, res FetchResult, dc, token, k string) erro
 		Index:     res.Index,
 		FetchedAt: time.Now(),
 		Waiter:    make(chan struct{}),
-		Expiry:    &cacheEntryExpiry{Key: key},
```

Comment:
@hashicorp/consul-cloud the only caller to `Cache.Prepopulate` is for AutoConfig of CA certs, so I wanted to double check this change with you.

I don't think the behaviour has changed here. When an item is added to the cache with Prepopulate it was never added to the `expiryHeap`. That means the value will stay in the cache for longer than 3 days if it does not change. I suspect that is ok in this case because we are also watching for changes to the certs, and any fetched change will populate the ExpiryHeap.

I removed the default value here, making it nil, because previously this was resulting in a call to `heap.Fix(0)`, which was technically not correct, but was also not a problem because nothing had changed.  With my refactor it became a problem because the call is now `Update(0, ttl)` which is not correct. It would refresh the wrong entry.

Reply:
> I suspect that is ok in this case because we are also watching for changes to the certs, and any fetched change will populate the ExpiryHeap.

https://github.com/hashicorp/consul/blob/6589cbbd0dc5058c5d2702616d1bc12ce2a531ad/agent/structs/connect_ca.go#L14
https://github.com/hashicorp/consul/blob/6589cbbd0dc5058c5d2702616d1bc12ce2a531ad/agent/cache-types/connect_ca_leaf.go#L261-L266

Having this in mind, the leaf cert should be touched well before 3days with the default `LeafCertTTL` which means we are fine either way. I am wondering what happens when the user configures `LeafCertTTL` to 10days. Does that mean that the cert would be removed from the cache?

Reply:
The prepopulated one would not be, but if they create another one with a fetch, then yes I think it could expire and be removed from the cache in that case.

Reply:
Even if it would be removed after say 6days, that wouldn't be a problem. The cache would refetch it and all is good I assume. It shouldn't be a problem because the expired cert still lives in `tlsConfigurator` and is being used for connections independently from it being present in the cache.